### PR TITLE
Improved tests and diagnostics

### PR DESCRIPTION
**Pull request by nohuhu**

### CODE REVIEW

**Comment by nohuhu** [comment]
I'm going to push this `1.30.0_01` release to CPAN to have it smoked before I can fix FreeBSD failures for good. One of the things we're missing is Postgres version that smokers are running; this PR takes care of that.

**Comment by TJC** [comment]
Given the other discussion around semver, maybe switch this PR to just be improved tests, and save the semver stuff for another time?

**Comment by nohuhu** [comment]
Already did with the latest commit, just forgot to update PR title. :)

**Comment by TJC** [comment]
Everything here looks pretty good -- just one thing occurred to me.

Quite a few tests are repeating the same work to get the current pg version. I think it would make more sense to query that from Test::PostgreSQL, so it's consistent and not repeated. I saw you added a pg_version method -- could that be used?

File: Changes [hunk 1]
```diff
@@ -2,6 +2,16 @@ Revision history for Perl extension Test::PostgreSQL.
...
+1.27_02
+    - Use pg_version attribute to better handle -b and -f psql switches
+      depending on installed PostgreSQL version
+    - Updated docs to reflect the changes
+    - Tests are passing with PostgreSQL 9.3+
+
+1.27_01
+    - pg_version attribute holds PostgreSQL version detected at startup
+    - Improved test diagnostics
+
```
Comment (TJC) [inline Changes]: If this is going into a new release, should the changes be under {{NEXT}} rather than the dev-release versions?

Comment (nohuhu) [reply Changes]: Right, these were probably remains of trial releases. I'll make the change.

File: t/09-run_psql.t [hunk 2]
```diff
@@ -14,13 +14,22 @@ plan tests => 3;
 
 ok defined($pg), "new instance created";
 
-eval {
-    $pg->run_psql(
+my @psql_command;
+
+# psql 9.6+ supports multiple -c commands
+if ( $pg->pg_version > 9.6 ) {
```
[the remark this answers is not in the document]
Comment (nohuhu) [reply t/09-run_psql.t]: It should indeed, thanks for spotting! I love it when more people look at the code. :)

**Comment by nohuhu** [comment]
All code review comments taken care of, CI is passing.

**Pull request merged**